app/src/main/java/com/example/nsc_events/screen/EventDetail.kt

```diff
@@ -7,7 +7,158 @@ import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.unit.dp
-import com.example.nsc_events.data.Event
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.Delete
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material3.Card
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.TopAppBar
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import com.example.nsc_events.R
+import com.example.nsc_events.Routes
+import com.example.nsc_events.data.Datasource
+import com.example.nsc_events.model.Event
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun EventDetailPage(navController: NavController, eventId: String) {
+    Scaffold(
+        modifier = Modifier
+            .fillMaxSize(),
+        topBar = {
+            /* Back arrow that navigates back to login page */
+            TopAppBar(
```

Comment:
We can move forward as is now I just want to make note that we should check in with @kkennedy-gh and make sure the way we're implementing the app bar is similar to his designs as IIRC he was the original app bar implementer. :) But this looks good! (It might be his app bar but I cant see code ownership in a PR unlike the IDE)